src/schema/users.ts

```diff
@@ -694,6 +706,48 @@ export const resolvers: IResolvers<any, Context> = {
       ctx: Context,
       info,
     ): Promise<Connection<GQLView>> => readHistoryResolver(args, ctx, info),
+    generateUniqueUsername: async (
+      _,
+      args: { name: string },
+      ctx: Context,
+    ): Promise<string> => {
+      const name = args.name.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
+
+      if (name.length === 0) {
+        return '';
+      }
+
+      const username = name.substring(0, 39);
+      let generatedUsernames = [username];
+
+      const generateRandomUsernames = (count: number) => {
+        for (let i = 0; i < count; i++) {
+          const random = randomInt(100);
+          const randomUsername = `${username.substring(0, 37)}${random}`;
+          generatedUsernames.push(randomUsername);
+        }
+      };
+
+      generateRandomUsernames(4);
+
+      const usernameChecks = await (async () => {
+        return ctx
+          .getRepository(User)
+          .findBy({ username: In(generatedUsernames) });
+      })();
+
+      for (const usernameCheck in usernameChecks) {
+        generatedUsernames = generatedUsernames.filter(
+          (item) => item !== usernameChecks[usernameCheck].username,
+        );
+      }
+
+      if (generatedUsernames.length === 0) {
```

Comment:
Non blocking request, could you perhaps add a log here (ctx.log.info) so we can keep track of these fails and see if we need to enhance the number of generated usernames.